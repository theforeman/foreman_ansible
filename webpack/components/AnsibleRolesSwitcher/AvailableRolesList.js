import React from 'react';

import { ListView, LoadingState } from 'patternfly-react';
import PaginationWrapper from 'foremanReact/components/Pagination/PaginationWrapper';

import AnsibleRole from './AnsibleRole';

const AvailableRolesList = ({
  unassignedRoles,
  pagination,
  itemCount,
  onListingChange,
  onAddRole,
  loading,
}) => (
  <ListView>
    <div className="sticky-pagination">
      <PaginationWrapper
        viewType="list"
        itemCount={itemCount}
        pagination={pagination}
        onChange={onListingChange}
        dropdownButtonId='available-ansible-roles-pagination-row-dropdown'
      />
    </div>
    <LoadingState loading={loading} >
      { unassignedRoles.map(role => <AnsibleRole key={role.id} role={role} icon='fa fa-plus-circle' onClick={onAddRole}/>) }
    </LoadingState>
  </ListView>
);

export default AvailableRolesList;
