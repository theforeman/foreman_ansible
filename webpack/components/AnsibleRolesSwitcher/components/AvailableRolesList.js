import React from 'react';
import PropTypes from 'prop-types';

import { ListView, LoadingState } from 'patternfly-react';
import PaginationWrapper from 'foremanReact/components/Pagination/PaginationWrapper';

import AnsibleRole from './AnsibleRole';

const AvailableRolesList = ({
  unassignedRoles,
  assignedRoles,
  pagination,
  itemCount,
  onListingChange,
  onAddRole,
  loading,
  resourceName,
  resourceId,
  variablesUrl,
}) => (
  <ListView>
    <div className="sticky-pagination">
      <PaginationWrapper
        viewType="list"
        itemCount={itemCount}
        pagination={pagination}
        onChange={onListingChange}
        dropdownButtonId="available-ansible-roles-pagination-row-dropdown"
      />
    </div>
    <LoadingState loading={loading}>
      {unassignedRoles.map(role => (
        <AnsibleRole
          key={role.id}
          role={role}
          icon="fa fa-plus-circle"
          onClick={onAddRole}
          resourceName={resourceName}
          resourceId={resourceId}
          variablesUrl={variablesUrl}
          assignedRoles={assignedRoles}
        />
      ))}
    </LoadingState>
  </ListView>
);
AvailableRolesList.propTypes = {
  unassignedRoles: PropTypes.arrayOf(PropTypes.object).isRequired,
  assignedRoles: PropTypes.array.isRequired,
  pagination: PropTypes.shape({
    page: PropTypes.number,
    perPage: PropTypes.number,
  }).isRequired,
  itemCount: PropTypes.number.isRequired,
  onListingChange: PropTypes.func.isRequired,
  onAddRole: PropTypes.func.isRequired,
  loading: PropTypes.bool.isRequired,
  resourceName: PropTypes.string.isRequired,
  resourceId: PropTypes.number,
  variablesUrl: PropTypes.string.isRequired,
};

AvailableRolesList.defaultProps = {
  resourceId: null,
};

export default AvailableRolesList;
