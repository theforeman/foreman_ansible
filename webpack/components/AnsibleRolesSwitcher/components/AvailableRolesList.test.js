import { testComponentSnapshotsWithFixtures } from '@theforeman/test';

import AvailableRolesList from './AvailableRolesList';

const noop = () => {};

const fixtures = {
  'should render': {
    unassignedRoles: [
      { id: 1, name: 'fake.role' },
      { id: 2, name: 'test.role' },
    ],
    pagination: { page: 1, perPage: 25 },
    itemCount: 2,
    onListingChange: noop,
    onAddRole: noop,
    loading: false,
    resourceName: 'Host',
    resourceId: 14,
    variablesUrl: '/hosts/14/variables',
    assignedRoles: [],
  },
};

describe('AvailableRolesList', () =>
  testComponentSnapshotsWithFixtures(AvailableRolesList, fixtures));
