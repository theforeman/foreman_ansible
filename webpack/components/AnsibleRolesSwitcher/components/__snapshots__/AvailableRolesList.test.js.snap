// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AvailableRolesList should render 1`] = `
<ListView
  className=""
>
  <div
    className="sticky-pagination"
  >
    <PaginationWrapper
      dropdownButtonId="available-ansible-roles-pagination-row-dropdown"
      itemCount={2}
      onChange={[Function]}
      pagination={
        Object {
          "page": 1,
          "perPage": 25,
        }
      }
      viewType="list"
    />
  </div>
  <LoadingState
    additionalClasses=""
    loading={false}
    loadingText="Loading"
    size="lg"
    timeout={300}
  >
    <AnsibleRole
      assignedRoles={Array []}
      icon="fa fa-plus-circle"
      key="1"
      onClick={[Function]}
      resourceId={14}
      resourceName="Host"
      role={
        Object {
          "id": 1,
          "name": "fake.role",
        }
      }
      variablesUrl="/hosts/14/variables"
    />
    <AnsibleRole
      assignedRoles={Array []}
      icon="fa fa-plus-circle"
      key="2"
      onClick={[Function]}
      resourceId={14}
      resourceName="Host"
      role={
        Object {
          "id": 2,
          "name": "test.role",
        }
      }
      variablesUrl="/hosts/14/variables"
    />
  </LoadingState>
</ListView>
`;
